common/plugins/org.jboss.tools.common.jdt.debug/src/org/jboss/tools/common/jdt/debug/tools/ToolsCore.java

```diff
@@ -94,8 +95,32 @@ public Object getMonitoredVM() {
 		return Tools.getInstance().invokeActiveVms(hostname);
 	}
 	
+	/**
+	 * This method was poorly named and is is instead desired to return whether a process
+	 * is able to be monitored or not. 
+	 * 
+	 * @param hostname
+	 * @param vmPid
+	 * @return
+	 * @throws ToolsCoreException
+	 */
+	@Deprecated
 	public static boolean processIsRunning(String hostname, int vmPid) throws ToolsCoreException {
-		return Tools.getInstance().getMonitoredVm(hostname, vmPid) != null;
+		return processIsMonitorable(hostname, vmPid);
+	}
+	
+	public static boolean processIsMonitorable(String hostname, int vmPid) throws ToolsCoreException {
+		// If these throw exceptions, we want them to be rethrown and logged
+		Object host = Tools.getInstance().invokeGetMonitoredHost(hostname);
```

Comment:
No... I consider those two critical errors. If the running VM cannot locate the monitored host (ie localhost) or is incapable of properly using the sun.jvmstat.monitor.VmIdentifier constructor (which is very simple), then I'd see that as a critical error that should be re-thrown and logged. 

If however we can access the monitored host, and can use the sun.jvmstat.monitor.VmIdentifier constructor without error, but an exception is thrown on invokeGetMonitoredVM (where most of the work is done, ie, where we ask the host to locate and connect to the process), then that should be caught and ignored, because that is an expected failure. 

Basically, the expected failures are inside the try/catch and ignored. The unexpected failures are outside of the try/catch, thrown up the stack, and logged.